src/stages/stages.controller.ts

```diff
@@ -2,20 +2,24 @@ import {
   Body,
   ConflictException,
   HttpStatus,
+  Inject,
+  NotFoundException,
   Post,
   UseGuards
 } from '@nestjs/common';
-import { ApiResponse, ApiBearerAuth } from '@nestjs/swagger';
+import { ApiBearerAuth, ApiResponse } from '@nestjs/swagger';
 import { BaseCrudController } from '~shared/controllers';
 import { Roles } from '~shared/decorators/roles.decorator';
 import { ApiException } from '~shared/errors';
 
 import { JwtAuthGuard, RolesGuard } from '../auth/guards';
+import { TracksService } from '../tracks/tracks.service';
 import { UserRole } from '../users/models/user.entity';
 import { CreateStageDto } from './models/dtos/create-stage.dto.ts';
 import { PagedListStageDto, StageDto } from './models/dtos/stage.dto.ts';
 import { Stage } from './models/stage.entity.ts';
 import { StagesService } from './stages.service';
+import { plainToClass } from 'class-transformer';
```

Comment:
Codacy found an issue: ['plainToClass' is defined but never used.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5781977)